packages/flutter/lib/src/widgets/framework.dart

```diff
@@ -460,6 +452,19 @@ abstract class Widget extends DiagnosticableTree {
 ///    widget, so that only the inner-most widget needs to be rebuilt when the
 ///    theme changes.
 ///
+///  * When trying to create a reusable piece of UI, prefer using a widget
+///    rather than a helper method. For example, if there was a function used to
+///    build a widget, a [State.setState] call would require Flutter to entirely
+///    rebuild the returned wrapping widget. If a [Widget] was used instead,
+///    Flutter would be able to efficiently re-render only those parts that
+///    really need to be updated. Even better, if the created widget is `const`,
+///    Flutter would short-circuit most of the rebuild work.
+///
+/// This video gives more explainations on why constant constructors are important
+/// and why a [Widget] is better than a helper method.
+///
+/// {@youtube 560 315 https://www.youtube.com/watch?v=wE7khGHVkYY}
```

Comment:
Did you mean to link a different video here? This one doesn't seem to talk about const constructors?

Reply:
Good catch, I meant to link this: [https://www.youtube.com/watch?v=IOyq-eTRhvo](https://www.youtube.com/watch?v=IOyq-eTRhvo)